### ISSUE TITLE
[New Features] I've added some new features in this PR

### ISSUE THREAD

**Issue by dpacchi** [ShekarMudaliyar/social_share#116]
Hi,

I've added some new features to this asset in [this](https://github.com/ShekarMudaliyar/social_share/pull/115) PR that hopefully will get approved, hopefully you'll find it useful!

All this features were tested in both iOS and Android

- Support for image background in both Facebook and Instagram stories
- Support for **video** background in both Facebook and Instagram stories
- Facebook App Id support in instagram stories (**required** starting december 2022)
- Clipboard sharing fixes 
- Example updated and fixed using the latest flutter release

![Screenshot_2022-11-03-12-41-37-006_com shekarmudaliyar social_share_example](https://user-images.githubusercontent.com/83594253/199712055-b05a08d6-1b7f-4765-baca-561b04ad601e.jpg)

**Comment by DennisAshford** [comment]
This sounds like an excellent (and soon to be required) update. Hopefully it is approved so we can implement the App ID.

**Comment by simplenotezy** [comment]
Yeah, let's get this merged so we can include the facebook app id for instagram stories starting from december already

**Comment by DennisAshford** [comment]
@dpacchi unfortunately looks like this package may be abandoned, which is a shame given how many people seem to use it from the ratings. I tried pointing to your branch(which seemed to work given the required App ID attribute), but I cannot compile/build on iOS or Android. Here is the error from XCode (note it works fine for the original package). Is your branch public? Or any idea where this may be coming from?

`2022-11-21 08:40:49.152 xcodebuild[24455:252580]  DVTPortal: Service '<DVTPortalViewDeveloperService: 0x7ff2032607d0; action='viewDeveloper'>' encountered an unexpected result code from the portal ('1100')
    2022-11-21 08:40:49.152 xcodebuild[24455:252580]  DVTPortal: Error:
    Error Domain=DVTPortalServiceErrorDomain Code=1100 "Your session has expired. Please log in." UserInfo={payload={
        creationTimestamp = "2022-11-21T13:40:49Z";
        httpCode = 200;
        protocolVersion = QH65B2;
        requestUrl = "https://developerservices2.apple.com/services/QH65B2/viewDeveloper.action";
        responseId = "f936065d-b15f-4c1f-9151-0282c387d779";
        resultCode = 1100;
        resultString = "Your session has expired. Please log in.";
        userLocale = "en_US";
        userString = "Your session has expired. Please log in.";
    }, NSLocalizedDescription=Your session has expired. Please log in.}`

**Comment by DennisAshford** [comment]
@dpacchi I am logged in, and it built fine with the older share package. Having similar issues with Android. Was curious if you had this issue pointing to your repo for the package?

**Comment by ShekarMudaliyar** [comment]
Apologies for the wait. I kept this project on hold as i was busy in some other work. 
I will be starting work on this from this week.
Also i wanted to rewrite this package as its too old to be maintained. Please email me if anyone wants to give me a hand in this project.

Thank you everyone for the contribution really appreciate it.😄

**Comment by DennisAshford** [comment]
@ShekarMudaliyar 🙌 🙌

I am not skillful enough to contribute code, but I will be contributing some coffee to the effort 😃

**Comment by dpacchi** [comment]
@ShekarMudaliyar nice to hear you! 
Rewriting this package from scratch could be a good idea, and that's what I have done with some of the features in my PR. 
Feel free to approve it, take parts of the code in it, or just take inspiration. 

This are the main features I've edited / rewritten

- Support for image background in both Facebook and Instagram stories
- Support for video background in both Facebook and Instagram stories
- Facebook App Id support in instagram stories (required starting december 2022)
- Clipboard sharing fixes
- Example updated and fixed using the latest flutter release  

I'd love to know what you decide to do with the rewriting of this package.
Ciao!

**Comment by DennisAshford** [comment]
Just FYI, if anyone runs into the error I mentioned above, a simple `Flutter Upgrade` did the trick for me. The newest package requires a newer version of Flutter than I had.

**Comment by arpitBhalla** [comment]
Is there any update?

**Comment by bhargav6744** [comment]
Any update on the same, i can help writing iOS Part if needed @ShekarMudaliyar
